### DISCUSSION TITLE
Custom Views and Widget Controllers

### DISCUSSION THREAD

**Discussion by Read2uk** [Laravel-Backpack/community-forum#1183] [Q&A (Help)]
Hi all,

I am trying to find the best way to use Custom Views and Widgets together.

1. These Widgets and chart are set in the setupListOperation, how can I hide them when I change the view? they still appearing in the other views, or have them display the custom view clause filter
![image](https://github.com/user-attachments/assets/58d0f783-cf71-47b9-9340-2cc9937299db)

2. How can I pass a parameter/filter to the Widget controller?

For example, when changing view I want to use the same ChartController but pass a filter to the query to pull the data
![image](https://github.com/user-attachments/assets/c480c8aa-196c-4155-8cf4-11c90499f325)

3. When I add a CRUD::addClause with a SQL filter in the CustomView function, its only applying the filter to the datatable, is there a smart way to also applying to the widgets and charts?

Regards

**Comment by karandatwani92** [comment]
Hey @Read2uk 

1. How to remove widgets:
```php
Widget::add('my_widget')
        ->to('before_content')
        ->type('card')
        ->content(null);

Widget::name('my_widget')->remove();
```

**Comment by karandatwani92** [comment]
2. You can catch filter parameters in setupListOperation `request('param')`; however, the standard chart widget does not pass any data to the chart controller.

I found a thread https://github.com/Laravel-Backpack/community-forum/discussions/532 on this. I wonder if you may find a custom solution that fits your needs.

**Comment by karandatwani92** [comment]
3. `CRUD::addClause()` is available only in CRUDController, which only applies to CRUD's model.
Widget Controllers are not bound to CRUD's specific model, so use any model directly inside it. It's just wapper on [Eloquent's Queries builder](https://laravel.com/docs/11.x/eloquent#building-queries).
